Sources/Perform/UIStoryboardSegue.swift

```diff
@@ -0,0 +1,18 @@
+import UIKit
+
+extension UIStoryboardSegue {
+  @nonobjc
+  public final func destinationViewController<Destination: UIViewController>(ofType type: Destination.Type) -> Destination? {
+    var queue: [UIViewController] = [destinationViewController]
+
+    while let viewController = queue.popLast() {
+      queue.insertContentsOf(viewController.childViewControllers, at: 0)
+
+      if let destination = viewController as? Destination {
+        return destination
+      }
+    }
```

Comment:
So to make sure I'm understanding this correctly, if I'm segueing to a navigation controller that contains my `type` controller, it will iterate over it until it finds it? No need for some `navigationController.viewControllers.first` stuff?! Cause that's awesome!